contracts/XVSVault/XVSVault.sol

```diff
@@ -0,0 +1,359 @@
+        }
+
+        totalAllocPoints[_rewardToken] = totalAllocPoints[_rewardToken].add(_allocPoint);
+
+        rewardTokenAmountsPerBlock[_rewardToken] = _rewardPerBlock;
+
+        poolInfo.push(
+            PoolInfo({
+                token: _token,
+                allocPoint: _allocPoint,
+                lastRewardBlock: block.number,
+                accRewardPerShare: 0
+            })
+        );
+
+        IXVSStore(xvsStore).setRewardToken(_rewardToken, true);
+    }
+
+    // Update the given pool's reward allocation point. Can only be called by the admin.
+    function set(
+        address _rewardToken,
+        uint256 _pid,
+        uint256 _allocPoint,
+        bool _withUpdate
+    ) public onlyAdmin {
+        if (_withUpdate) {
+            massUpdatePools(_rewardToken);
+        }
+
+        PoolInfo[] storage poolInfo = poolInfos[_rewardToken];
+        totalAllocPoints[_rewardToken] = totalAllocPoints[_rewardToken].sub(poolInfo[_pid].allocPoint).add(
+            _allocPoint
+        );
+        poolInfo[_pid].allocPoint = _allocPoint;
+    }
+
+    // Update the given reward token's amount per block
+    function setRewardAmountPerBlock(
+        address _rewardToken,
+        uint256 _rewardAmount
+    ) public onlyAdmin {
+        rewardTokenAmountsPerBlock[_rewardToken] = _rewardAmount;
+    }
+
+    // Update the given reward token's amount per block
+    function setWithdrawalLockingPeriod(
+        uint256 _newPeriod
+    ) public onlyAdmin {
+        require(_newPeriod > 0, "Invalid new locking period");
+        lockPeriod = _newPeriod;
+    }
+
+    /**
+     * @notice Deposit XVSVault for XVS allocation
+     * @param _rewardToken The Reward Token Address
+     * @param _pid The Pool Index
+     * @param _amount The amount to deposit to vault
+     */
+    function deposit(address _rewardToken, uint256 _pid, uint256 _amount) public nonReentrant {
+        PoolInfo storage pool = poolInfos[_rewardToken][_pid];
+        UserInfo storage user = userInfos[_rewardToken][_pid][msg.sender];
+        updatePool(_rewardToken, _pid);
+        if (user.amount > 0) {
+            uint256 pending =
+                user.amount.mul(pool.accRewardPerShare).div(1e12).sub(
+                    user.rewardDebt
+                );
+            IXVSStore(xvsStore).safeRewardTransfer(_rewardToken, msg.sender, pending);
+        }
+        pool.token.transferFrom(
+            address(msg.sender),
+            address(this),
+            _amount
+        );
+        user.amount = user.amount.add(_amount);
+        user.rewardDebt = user.amount.mul(pool.accRewardPerShare).div(1e12);
+        emit Deposit(msg.sender, _rewardToken, _pid, _amount);
+    }
+
+    /**
+     * @notice Execute withdrawal to XVSVault for XVS allocation
+     * @param _rewardToken The Reward Token Address
+     * @param _pid The Pool Index
+     */
+    function ExecuteWithdrawal(address _rewardToken, uint256 _pid) public nonReentrant {
```

Comment:
Why does it start with an uppercase letter though, any particular reason for this?

```suggestion
    function executeWithdrawal(address _rewardToken, uint256 _pid) public nonReentrant {
```